lib/resolvers/profile.js

```diff
@@ -61,5 +61,28 @@ module.exports = ({ models }) => ({ action, data, id }, transaction) => {
       .then(() => Profile.query(transaction).findById(data.target));
   }
 
+  if (action === 'update' && data.email) {
+    let oldEmail = '';
+
+    // attempt to update email in keycloak
+    return Promise.resolve()
+      .then(() => keycloak.grantToken())
+      .then(accessToken => {
+        return Profile.query(transaction).findById(id)
+          .then(profile => {
+            oldEmail = profile.email;
+            return { id: profile.userId, email: data.email };
+          })
+          .then(user => keycloak.updateUser({ accessToken, user }));
+      })
+      .then(() => resolver({ Model: models.Profile, action, data, id }, transaction))
+      .then(profile => emailer.sendEmail({ template: 'change-email', ...profile, oldEmail }))
+      .catch(err => {
+        const error = new Error('There was a problem updating the user in keycloak');
```

Comment:
If you're going to pin the error on keycloak then it would be best to move this `catch` block up so it's only catching keycloak errors.

This will attribute any error in the database update of emailer to keycloak as well.